homeassistant/components/camera/demo.py

```diff
@@ -38,3 +45,30 @@ def camera_image(self):
     def name(self):
         """Return the name of this camera."""
         return self._name
+
+    @property
+    def get_motion_detection_status(self):
+        """Camera Motion Detection Status."""
+        return self._motion_status
+
+    @asyncio.coroutine
+    def async_enable_motion_detect(self):
+        """Camera arm."""
+        self._motion_status = MOTION_ENABLED
+        self.hass.async_add_job(self.async_update_ha_state())
+        self.hass.async_add_job(self.async_update())
+
+    @asyncio.coroutine
+    def async_disable_motion_detect(self):
+        """Camera disarm."""
+        self._motion_status = MOTION_DISABLED
+        self.hass.async_add_job(self.async_update_ha_state())
+        self.hass.async_add_job(self.async_update())
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Perform the I/O operation with camera."""
+        if self._motion_status == MOTION_ENABLED:
+            _LOGGER.info("Enable Motion detection for this camera here")
```

Comment:
I have added the should_poll property override and removed the async_update code from the demo.py